### ISSUE TITLE
November 2023 Endgame

### ISSUE THREAD

**Issue by justschen** [microsoft/vscode#198892]
- *11/27* Code freeze for the endgame
- *12/1* [Endgame done](https://github.com/Microsoft/vscode/wiki/Development-Process#inside-an-iteration)
- *12/6* Expected release date (this may change)

##### Monday

- [ ] Check that all queries in this issue use the current milestone @justschen 
- [ ] Run [OSS tool](https://github.com/microsoft/vscode-build-tools/tree/main/distro-tools) @justschen 
- [ ] Update links in the [Endgame notebooks](https://github.com/microsoft/vscode/blob/main/.vscode/notebooks/endgame.github-issues) to point to new milestone @justschen 
- [ ] Code freeze at 5pm PT
- [ ] Ensure we have a green build on all platforms at 5pm PT
- [ ] [🔖All closed feature-requests](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+milestone%3A%22${milestone}%22+label%3Afeature-request+-label%3Averification-needed+-label%3Aon-testplan+-label%3Averified+-label%3A*duplicate+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger) either have a `verification-needed` or `on-testplan` label
- [ ] Create test plan items following the template [here](https://github.com/microsoft/vscode/wiki/Writing-Test-Plan-Items) by 6pm PT
  - [ ] Remind the team that TPIs should be written so that anyone can test. If this is not feasible, then TPI authors should assign specific testers **endgame champion**
- [ ] Update your availability for testing here - https://tools.code.visualstudio.com/team-manifest **team**
  - [ ] Update availability of testers in vacation (Check OOF section in the Internal Backlog Plan). Double check N/A testers. **endgame champion**
  - [ ] Remind team that in the event of sickness or overload with TPIs, to inform the endgame champ ASAP so items can be reassigned
- [ ] Remind team to go through their fixed issues for the milestone and update repro steps for issues which require more detailed instructions.

##### Tuesday

- [ ] Test plan items assigned (using https://tools.code.visualstudio.com/test-plan-items)
  - Run the tool multiple times to balance load if test items come in later and assignments are already made
  - [Assigned to you](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+assignee%3A%40me++label%3Atestplan-item+)
- [ ] Test build starts at 7am CET
- [ ] Test plan ready by 8am CET
- [ ] Remind the team about the priorities
  - Tuesday should be dedicated _exclusively_ to testing activities. Our goal is to ensure the completion of all Test Plan Items (TPIs) and subsequently proceed with the verification phase. Fixes or commits should be refrained from unless there are exceptional circumstances such as blocked TPIs or build-related issues. On Tuesday Redmond EOD, only blocked TPIs should remain open with a corresponding label `blocked` and status update comment in the issue.
  - Redmond members commence testing the TPIs authored by Zurich members, enabling them to address any feedback on the following day.
- [ ] [🔖Testing](https://github.com/issues?q=is%3Aopen+is%3Aissue+label%3Atestplan-item+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger)
- [ ] [🔖Verification needed](https://github.com/issues?q=is%3Aissue+milestone%3A%22${milestone}%22+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Averification-needed+-label%3Averified+is%3Aclosed+-label%3Aon-testplan)

##### Wednesday

- [ ] [🔖Testing](https://github.com/issues?q=is%3Aopen+is%3Aissue+label%3Atestplan-item+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger)
  - These remaining TPIs should be the ones that are currently blocked. Discuss during the standup and redistrubute assignments based on the TPI owner and the test coverage. For instance, if a TPI is owned by a member from Zurich and has not undergone sufficient testing, it will be reassigned to one of the Zurich team members.
- [ ] Remind team members to triage issues found in testing and assign major issues that they intend to fix to the current milestone
- [ ] [🔖Verification needed](https://github.com/issues?q=is%3Aissue+milestone%3A%22${milestone}%22+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Averification-needed+-label%3Averified+is%3Aclosed+-label%3Aon-testplan)

##### Thursday

- [ ] Make sure to have the `insider` build **endgame champion**
- [ ] Pause scheduled `insider` builds (includes vscode.dev publishing) **endgame champion**
- [ ] Fixing (self-assigned, milestone assigned)
  - Increased scrutiny sets in due to testing being completed. Fixes pose a much higher risk
  - Move issues to the next month that can be deferred
- [ ] [🔖Verification needed](https://github.com/issues?q=is%3Aissue+milestone%3A%22${milestone}%22+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Averification-needed+-label%3Averified+is%3Aclosed+-label%3Aon-testplan)
- [ ] [🔖Verification](https://github.com/issues?q=is%3Aissue+milestone%3A%22${milestone}%22+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Abug+-label%3Aduplicate+-label%3A*duplicate+-label%3Ainvalid+-label%3Aas-designed+-label%3Aerror-telemetry+is%3Aclosed+-label%3Averified+-label%3Aon-testplan)
- [ ] By Thursday EOD (Redmond), branch from `main` and release **endgame champion**
  - [ ] Branch following repositories to `release/<x.y>`
    - [ ] vscode
    - [ ] vscode-distro
    - [ ] vscode-dev
  - [ ] Announce `main` is open for business and all issues on the current iteration are candidates and that the candidate release process is to be followed.
  - [ ] Build and manually release Insider from release/<x.y> branch for Code
  - [ ] Build and manually release Insider from release/<x.y> branch for vscode.dev (if there were vscode.dev candidates)
- [ ] Build a `stable` build from release/<x.y> branch to ensure stable build is green **endgame champion**
- [ ] Bump up the version in `package.json` on `main` - **endgame champion**
- [ ] [Create next milestone](https://github.com/microsoft/vscode-internalbacklog/milestones/new) and ensure that it has a due date. The created milestone and its due date will be automatically synced across our repos **endgame champion**

##### Friday

- [ ] Only candidate issues are open and assigned to [🔖milestone](https://github.com/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22${milestone}%22+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger)
- [ ] All issues [🔖verified](https://github.com/issues?q=is%3Aissue+milestone%3A%22${milestone}%22+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger+-label%3Aiteration-plan+-label%3Aendgame-plan+-label%3Atestplan-item+label%3Abug+-label%3Aduplicate+-label%3A*duplicate+-label%3Ainvalid+-label%3Aas-designed+-label%3Aerror-telemetry+is%3Aclosed+-label%3Averified+-label%3Aon-testplan)
- [ ] All open PRs on the milestone [🔖merged or deferred](https://github.com/issues?q=is%3Apr+milestone%3A%22${milestone}%22+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger+is%3Aopen)
- [ ] Fixing (PR + review required - major bugs only - to be discussed in stand-up meeting, labeled as `candidate`)
- [ ] Localization: Run [Update VS Code Branch](https://monacotools.visualstudio.com/Monaco/_build?definitionId=369) build with `release/*` as the VS Code Branch parameter (it's the default so you shouldn't have to change anything) **endgame champion**
- Satellite modules/npm packages ready, version updated, smoke tested
  - [ ] yo generator **@aeschli**
  - [ ] js-debug **@connor4312**
  - [ ] Debug Adapter Protocol **@connor4312**
  - [ ] vscode-debugadapter-node **@connor4312**
  - [ ] vscode-test-web **@aeschli**
  - [ ] vsce **@joaomoreno**
- [ ] All release notes updated
  - release notes are collected in a file named _`v<Major>_<Minor>.md`\_ in this [repo directory](https://github.com/Microsoft/vscode-docs/blob/vnext/release-notes/)
  - Release notes screen recording guidance: https://microsoft-my.sharepoint.com/:p:/p/buhollan/Ecg_j89jzlVGtfqWGy7SQBEB_OsuLPZ65jjKRVlo_OdlgA
- [ ] Acknowledge pull requests in release notes. We acknowledge PRs from outside the team. We have improved [the tooling](https://tools.code.visualstudio.com/acknowledgement) so that the endgame champion can generate the pull request acknowledgment for all repositories at once. **endgame champion**
  - [ ] `debug-adapter-protocol`, `inno-updater`, `jsonc-parser`, `language-server-protocol`, `lsif-node`, `vscode`, `vscode-codicons`, `vscode-css-languageservice`, `vscode-debugadapter-node`, `vscode-dev-containers`, `vscode-docs`, `vscode-emmet-helper`, `vscode-eslint`, `vscode-extension-samples`, `vscode-generator-code`, `vscode-hexeditor`, `vscode-html-languageservice`, `vscode-js-debug`, `vscode-js-debug-companion`, `vscode-js-profile-visualizer`, `vscode-jshint`, `vscode-json-languageservice`, `vscode-languageserver-node`, `vscode-livepreview`, `vscode-loader`, `vscode-lsif-extension`, `vscode-node-debug`, `vscode-node-debug2`, `vscode-pull-request-github`, `vscode-recipes`, `vscode-references-view`, `vscode-textmate`, `vscode-vsce`, `vscode-pylint`, `vscode-black-formatter`, `vscode-isort`, `lsprotocol`, `vscode-flake8`, `vscode-autopep8`, `vscode-python-tools-extension-template`, `vscode-mypy`, `vscode-python-debugger`
- [ ] Acknowledge [issue trackers](https://github.com/microsoft/vscode-internalbacklog/wiki/Community-Triage---Credits) from the community **@chrmarti**
- When done fixing/verifying and there are changes since last build at the end of day PT
  - [ ] Build and manually release Insider from release/<x.y> for Code **endgame champion**
  - [ ] Build and manually release Insider from release/<x.y> for vscode.dev (if there were vscode.dev candidates) **endgame champion**

##### Friday/Monday

- [ ] Polish release notes **redmond**
- [ ] Fixing (only critical bugs - no string changes)
- [ ] Schedule an endgame restrospective with with endgame champion, endgame buddy, and next endgame champion for this week. [Retrospective template](https://github.com/microsoft/vscode-internalbacklog/blob/main/.github/ISSUE_TEMPLATE/endgame-retrospective.md#L14)
- [ ] Decide whether a [Patch Tuesday release](https://github.com/microsoft/vscode-internalbacklog/wiki/Handling-MSRC-Cases#msrc--patch-tuesday-process) will happen:
  - Check the internal iteration plan for scheduled security issues (MSRCs) to determine if a Patch Tuesday release is needed.
  - If no Patch Tuesday release is planned, [let the Updates team know of that fact](https://github.com/microsoft/vscode-internalbacklog/wiki/Patch-Tuesday-E-Mail-Template) ([CSEClientPatching@microsoft.com](mailto:CSEClientPatching@microsoft.com)) **endgame champion**
  - Otherwise, if a Patch Tuesday release is necessary, start off a [Patch Tuesday Release plan](https://github.com/microsoft/vscode-internalbacklog/wiki/Handling-MSRC-Cases#are-you-endgame-champbuddy) **endgame champion**
- [ ] Remind the team: if they are going to be OOF for more than five days during the next iteration, assign someone to look out for critical issues in their feature areas and fix them if necessary. This helps with bug identification and fixing for recovery releases. **endgame champion**

##### Monday - Wednesday

- [ ] [🔖milestone issues](https://github.com/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22${milestone}%22+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger)
- [ ] [🔖candidate issues](https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3Acandidate+repo:microsoft/monaco-editor%20repo:microsoft/vscode%20repo:microsoft/vscode-anycode%20repo:microsoft/vscode-copilot%20repo:microsoft/vscode-copilot-release%20repo:microsoft/vscode-dev%20repo:microsoft/vscode-dev-chrome-launcher%20repo:microsoft/vscode-emmet-helper%20repo:microsoft/vscode-extension-telemetry%20repo:microsoft/vscode-github-issue-notebooks%20repo:microsoft/vscode-hexeditor%20repo:microsoft/vscode-internalbacklog%20repo:microsoft/vscode-js-debug%20repo:microsoft/vscode-jupyter%20repo:microsoft/vscode-jupyter-internal%20repo:microsoft/vscode-l10n%20repo:microsoft/vscode-livepreview%20repo:microsoft/vscode-pull-request-github%20repo:microsoft/vscode-python%20repo:microsoft/vscode-python-debugger%20repo:microsoft/vscode-references-view%20repo:microsoft/vscode-remote-release%20repo:microsoft/vscode-remote-repositories-github%20repo:microsoft/vscode-remote-tunnels%20repo:microsoft/vscode-remotehub%20repo:microsoft/vscode-settings-sync-server%20repo:microsoft/vscode-unpkg%20repo:microsoft/vscode-vsce%20repo:microsoft/vscode-pylint%20repo:microsoft/vscode-black-formatter%20repo:microsoft/vscode-isort%20repo:microsoft/lsprotocol%20repo:microsoft/vscode-flake8%20repo:microsoft/vscode-autopep8%20repo:microsoft/vscode-python-tools-extension-template%20repo:microsoft/vscode-mypy%20repo:microsoft/vscode-python-debugger)
- [ ] Polish release notes **redmond**
- [ ] Cherry-pick hand-picked and reviewed changes to `release/<x.y>` **endgame champion**
- [ ] Build `Insider` from `release/<x.y>` **endgame champion**
- [ ] Manually release `Insider` **endgame champion**
- [ ] Build stable for all platforms as new candidate issues come in **endgame champion**
- [ ] Documentation updated
- [ ] Run `scripts/test-documentation.sh|bat` and add file or fix issues if there are new colors that are not documented. **endgame champion**

> **Note:** The `Insiders` build needs to be in the wild for 24 hours before we can enter the last phase of the endgame. **endgame champion**

##### Wednesday/Thursday - Expected release day (this may change)

- [ ] Build stable for all platforms **endgame champion**
- [ ] Build but don't release an insider build based on `main` for all platforms **endgame champion**
- [ ] [Sanity Testing](https://github.com/microsoft/vscode/wiki/Sanity-Check)
  - [ ] Windows 64 bit **owner**
    - [ ] System Installer
    - [ ] User Installer
    - [ ] Archive
    - [ ] Server
    - [ ] CLI
  - [ ] Windows ARM64 **owner**
    - [ ] System Installer
    - [ ] User Installer
    - [ ] Archive
    - [ ] CLI
  - [ ] macOS Intel **owner**
    - [ ] Archive
    - [ ] Universal Archive
    - [ ] Server
    - [ ] CLI
  - [ ] macOS ARM64 **owner**
    - [ ] Archive
    - [ ] Universal Archive
    - [ ] Server
    - [ ] CLI
  - [ ] Linux x64 **owner**
    - [ ] Archive
    - [ ] DEB
    - [ ] RPM
    - [ ] Snap
  - [ ] Linux Server **owner** _(can be tested from any OS with Docker)_
    - [ ] x64
    - [ ] ARM32
    - [ ] ARM64
    - [ ] x64 Alpine
    - [ ] ARM64 Alpine
  - [ ] Linux CLI **owner** _(can be tested from any OS with Docker)_
    - [ ] x64
    - [ ] ARM32
    - [ ] ARM64
    - [ ] x64 Alpine
    - [ ] ARM64 Alpine
- [ ] Publish website **@gregvanl**
- [ ] Publish to stable **endgame champion**
- [ ] Trigger the [vscode.dev Deploy pipeline](https://dev.azure.com/monacotools/Monaco/_build?definitionId=276) for `release/x.y` for the `prod` deployment target. Note that there are four deploy approvals needed - one for overall and one per each of the three service regions **endgame champion**
  - [ ] Request approval from another team member at the necessary step to deploy the vscode.dev build. **endgame champion**
- [ ] Create an official release **endgame champion**
  - [ ] Create a tag (make sure you pull the release branch first): `git tag <x.y.z>`
  - [ ] Push the tag: `git push origin <x.y.z>`
  - [ ] Create a GitHub release: [Open the GitHub tags](https://github.com/microsoft/vscode/tags), and click far right `... > Create Release`. Use the correct title and description from our release notes. Also change the relative links for the key highlight list items to absolute links [Example](https://github.com/microsoft/vscode/releases/tag/1.39.0)
- [ ] Twitter announcement **@chrisdias**
- [ ] Enable scheduled `insider` builds **endgame champion**
- [ ] Release insiders based on `main` **endgame champion**
- [ ] [Publish @types/vscode](https://github.com/microsoft/vscode/wiki/Publish-vscode-types) **endgame champion**
- [ ] Close the milestone on [GitHub](https://github.com/microsoft/vscode/milestones) **endgame champion**

**Comment by justschen** [comment]
SORRY for the pings - closing as its in the wrong repo.
